R/waapwlspetpeese.R

```diff
@@ -0,0 +1,610 @@
+  }
+
+  # PET-PEESE estimates
+  if (type == "petPeese") {
+    if (is.null(estimates[["petPeese"]]) && options[["estimatesPetPeese"]]) {
+      petPeese <- createJaspTable(title = gettext("PET-PEESE Regression Estimates"))
+      petPeese$position  <- 2
+      petPeese$dependOn(c("estimatesPetPeese"))
+      estimates[["petPeese"]] <- petPeese
+      petPeese <- .wwppFillPetPeese(jaspResults, petPeese, models, options)
+    }
+  }
+
+  # heterogeneity estimates
+  if (is.null(estimates[["heterogeneity"]]) && options[["estimatesSigma"]]) {
+    heterogeneity <- createJaspTable(title = gettext("Multiplicative Heterogeneity Estimates"))
+    heterogeneity$position <- 3
+    heterogeneity$dependOn(c("estimatesSigma"))
+    estimates[["heterogeneity"]] <- heterogeneity
+    heterogeneity <- .wwppFillHeterogeneity(jaspResults, heterogeneity, models, options, type)
+  }
+
+  return()
+}
+# create the plots
+.wwppRegressionPlot          <- function(jaspResults, dataset, options, type = "pet") {
+
+  if (!is.null(jaspResults[[paste0(type, "Regression")]])) {
+    return()
+  } else {
+    plotRegression <- createJaspPlot(
+      title  = gettextf("Estimated %1$s Regression", toupper(type)),
+      width  = 500,
+      height = 400)
+    plotRegression$dependOn(c(.wwppDependencies, switch(type, "pet" = "regressionPet", "peese" = "regressionPeese")))
+    plotRegression$position <- switch(type, "pet" = 5, "peese" = 6)
+    jaspResults[[paste0(type, "Regression")]] <- plotRegression
+  }
+
+  if (!.wwppCheckReady(options))
+    return()
+
+
+  # get the fit
+  fit   <- attr(jaspResults[["models"]]$object[[type]], "fit")
+  fitEs <- attr(jaspResults[["models"]]$object[[type]], "y")
+  fitSe <- attr(jaspResults[["models"]]$object[[type]], "se")
+
+  # get the regression line prediction
+  fitSeRange    <- range(jaspGraphs::getPrettyAxisBreaks(c(0, fitSe)))
+  fitSeSequence <- seq(fitSeRange[1], fitSeRange[2], length.out = 101)
+  fitPrediction <- predict(fit, newdata = data.frame(se = fitSeSequence), interval = "confidence")
+
+  if (options[["measures"]] == "correlation")
+    yLabel <- switch(
+      options[["muTransform"]],
+      "cohensD"  = gettext("Cohen's d"),
+      "fishersZ" = gettext("Fisher's z"))
+  else
+    yLabel <- gettext("Effect Size")
+
+  # make the plot happen
+  plot <- ggplot2::ggplot()
+  # mean
+  plot <- plot + ggplot2::geom_polygon(
+    ggplot2::aes(
+      x = c(fitSeSequence, rev(fitSeSequence)),
+      y = c(fitPrediction[,"lwr"], rev(fitPrediction[,"upr"]))
+    ),
+    fill = "grey80")
+  # CI
+  plot <- plot +ggplot2::geom_path(
+    ggplot2::aes(
+      x = fitSeSequence,
+      y = fitPrediction[,"fit"]
+    ),
+    size = 1.25)
+
+  plot <- plot + ggplot2::scale_x_continuous(
+    gettext("Standard Error"),
+    breaks = jaspGraphs::getPrettyAxisBreaks(c(0, fitSe)),
+    limits = range(jaspGraphs::getPrettyAxisBreaks(c(0, fitSe))),
```

Comment:
What is now limits is equal to `fitSeRange` right? Maybe just make the breaks and let limits be `range(xBreaks)` and breaks `xBreaks`.

Reply:
true, changed